core/src/test/java/com/graphhopper/routing/DirectedBidirectionalDijkstraTest.java

```diff
@@ -0,0 +1,536 @@
+        assertPath(calcPath(0, 6, left0, left6, createWeighting(40)), 43.6, 60, 43600, nodes(0, 7, 8, 9, 6, 1, 6));
+    }
+
+    @Test
+    @Repeat(times = 10)
+    public void compare_standard_dijkstra() {
+        compare_with_dijkstra(weighting);
+    }
+
+    @Test
+    @Repeat(times = 10)
+    public void compare_standard_dijkstra_finite_uturn_costs() {
+        compare_with_dijkstra(createWeighting(40));
+    }
+
+    private void compare_with_dijkstra(Weighting w) {
+        // if we do not use start/target edge restrictions we should get the same result as with Dijkstra.
+        // basically this test should cover all kinds of interesting cases except the ones where we restrict the
+        // start/target edges.
+        final long seed = System.nanoTime();
+        final int numQueries = 1000;
+        System.out.println("compare_standard_dijkstra seed: " + seed);
+
+        Random rnd = new Random(seed);
+        int numNodes = 100;
+        GHUtility.buildRandomGraph(graph, rnd, numNodes, 2.2, true, true, encoder.getAverageSpeedEnc(), 0.7, 0.8, 0.8);
+        GHUtility.addRandomTurnCosts(graph, seed, encoder, maxTurnCosts, turnCostExtension);
+
+        long numStrictViolations = 0;
+        for (int i = 0; i < numQueries; i++) {
+            int source = rnd.nextInt(numNodes);
+            int target = rnd.nextInt(numNodes);
+            Path dijkstraPath = new Dijkstra(graph, w, TraversalMode.EDGE_BASED).calcPath(source, target);
+            Path path = calcPath(source, target, ANY_EDGE, ANY_EDGE, w);
+            assertEquals("dijkstra found/did not find a path, from: " + source + ", to: " + target, dijkstraPath.isFound(), path.isFound());
+            assertEquals("weight does not match dijkstra, from: " + source + ", to: " + target, dijkstraPath.getWeight(), path.getWeight(), 1.e-6);
+            // we do not do a strict check because there can be ambiguity, for example when there are zero weight loops.
+            // however, when there are too many deviations we fail
+            if (
+                    Math.abs(dijkstraPath.getDistance() - path.getDistance()) > 1.e-6
+                            || Math.abs(dijkstraPath.getTime() - path.getTime()) > 10
+                            || !dijkstraPath.calcNodes().equals(path.calcNodes())) {
+                numStrictViolations++;
+            }
+        }
+        if (numStrictViolations > Math.max(1, 0.05 * numQueries)) {
+            fail("Too many strict violations: " + numStrictViolations + " / " + numQueries);
+        }
+    }
+
+    @Test
+    @Ignore("todo: do we require running the algo with an additional edge filter ?")
+    public void worksWithEdgeFilter() {
+        // todo:
+        // e.g. block area is implemented using infinite weighting
+        // when an edge filter is used need to think about what should happen if the filter filters out the
+        // given in out edges etc.
```

Comment:
Interesting and yes, if we do a block area and restrict the edge (with infinite weight) then we should get "connection not found".

Reply:
Yes, this is already covered. I added a test for this and removed the todo.